the-collab-lab/tcl-67-smart-shopping-list
# Rb ts improve list view 52

## Description

List view and search bar improvements.

Todo:

- [x] Hide search bar if no items in that list.
- [x] Search bar clear input button should be in line with search bar
- [x] Search copy:
- [x] Search your shopping list
- [x] Search for items in your shopping list
- [x] Clear instead of just X
- [x] Hitting enter in search bar should not refresh page
- [x] remove: Hello from the /list page!
- [x] Add item form should be above list in list page
- [x] Share list form should be under list in list view, or to bottom right
- [x] Add list name to list view
- [x] 'No items in list, please add items' message when the list view is empty got removed somehow. Please fix

## Related Issue

closes #52 

## Type of Changes

Enhancements/Bug Fixes

## Updates

### Before
<img width="353" alt="old1" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/14006060/3ee3d5b4-bedd-4715-ae84-95feada4f9fb">

<img width="538" alt="old2" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/14006060/88701bc2-e147-41cf-a768-f57f6c682d6d">


### After
<img width="437" alt="new1" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/14006060/6ce22475-05e1-4424-8c0d-1c818a62f6fd">

<img width="537" alt="new2" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/14006060/2637d05f-6ea7-47fc-9d09-3891828981dd">


## Testing Steps / QA Criteria

The majority of these changes were aesthetic: 

1) Manage list and List View are now combined.
2) Copy has been updated for buttons on list page and list names on home page
3) Removal of manage list nav link at the bottom of the page
4) Properly displays correct messages when list is empty on list page.
5) Enter key while focus is on search item field no longer clears the search or hard refreshes the page.
